src/diffusers/utils/hub_utils.py

```diff
@@ -271,8 +272,23 @@ def move_cache(old_cache_dir: Optional[str] = None, new_cache_dir: Optional[str]
 def _add_variant(weights_name: str, variant: Optional[str] = None) -> str:
     if variant is not None:
         splits = weights_name.split(".")
-        split_index = -2 if weights_name.endswith(".index.json") else -1
-        splits = splits[:-split_index] + [variant] + splits[-split_index:]
+        # This if is for allowing to load sharded checkpoints with variants with the legacy format.
+        # Example: diffusion_pytorch_model-00001-of-00002.fp16.index.json
+        # Treatment of the legacy format.
+        if variant in weights_name and weights_name.endswith(".index.json"):
+            if weights_name.endswith(".index.json"):
+                if ".cache" in weights_name in weights_name:
```

Comment:
can you explain this?

Reply:
It was an oversight on my part. Shouldn't be concerning now.